polaris-react/src/components/Box/Box.tsx

```diff
@@ -207,24 +190,29 @@ export const Box = forwardRef<HTMLElement, BoxProps>(
     },
     ref,
   ) => {
+    // eslint-disable-next-line no-nested-ternary
+    const borderStyleValue = borderStyle
+      ? borderStyle
+      : borderColor || borderWidth
+      ? 'solid'
+      : undefined;
+
+    // eslint-disable-next-line no-nested-ternary
+    const outlineStyleValue = outlineStyle
+      ? outlineStyle
+      : borderColor || borderWidth
```

Comment:
Is this supposed to be checking for `outlineColor || outlineWidth`?

```suggestion
      : outlineColor || outlineWidth
```